### DISCUSSION TITLE
Cross References Syntax and cross-compatibility

### DISCUSSION THREAD

**Discussion by connorp** [quarto-dev/quarto-cli#393] [Q&A] [answered]
Hey all, I just stumbled upon Quarto and it looks super cool. I'm pumped! I wanted to ask briefly about how Cross References are handled. 

At the moment, there are (at least) two syntax forms of cross-references in pandoc-manipulated documents: `pandoc-crossref` and `bookdown`. I did notice that the plan was to [transition `bookdown` to use `pandoc-crossref`-style references](https://github.com/rstudio/bookdown/issues/811#issuecomment-827020479), but Quarto has made its own slight change (using `-` rather than `:`). Not to say that the motivation for this change isn't worthwhile, but I'm generally wary of conflicting syntax (as I'm sure you are as well). Are there any plans for any of the following? 

- Backwards compatibility mode in Quarto for existing documents that use `#fig:`
- Changes to `pandoc-crossref` that you have discussed with those maintainers to align the syntax? 

Under the hood, I'm also asking, is Quarto using `pandoc-crossref` (or a modified version) to do the referencing, or is it new code from whole cloth? If the latter, is it accessible and usable standalone? Not that I want to steer people away from Quarto, but since this is (I think) one of the few syntax enhancements atop pandoc that has to do purely with the markup, not with the code integration/literate programming, it would be nice if this were usable on its own with pandoc. Ideally, I'm hoping we can land on a standard syntax across (pandoc-related) packages, so that if you move a markdown document between them, some things might fail to render when a feature isn't supported, but at least there are no full conflicts.

**Comment by jjallaire** [comment, marked as answer]
Quarto does in fact use syntax very similar to `pandoc-crossref`, albeit with a few important differences:

1.  As you noted, Quarto uses the prefix `#fig-` rather than `#fig:` (this is more compatible with Jupyter notebook [cell ids](https://jupyter.org/enhancement-proposals/62-cell-id/cell-id.html), which don't allow `:`). 
2.  Quarto is able to reference raw HTML and LaTeX figures and tables (which are often produced by executable code blocks).
3.  Quarto has support for referencing theorems and proofs (and related types).

As such, this is a ground up implementation of rather than a direct re-use of `pandoc-crossref`

This is implemented as a set of Lua filters, so it is at least in-principle something that could be used by other systems. That said, crossref systems can't really be implemented without a stronger notion of both figures and figure panel layout (both `bookdown` and `pandoc-crossref` have figure/layout systems embedded in them). Those features could also in-principle be bundled together as a suite of Lua filters but we haven't done that work yet.

**Comment by kchousos** [comment]
Any update on this? Is there any progress in compatibility, or a smooth way to use Quarto crossrefs as standalone lua filters?
